core/market/decentralized/src/db/model/agreement.rs

```diff
@@ -111,6 +113,53 @@ impl Agreement {
             committed_signature: None,
         }
     }
+
+    pub fn into_client(self) -> Result<ClientAgreement, ErrorMessage> {
+        let demand_properties = serde_json::from_str(&self.demand_properties)
+            .map_err(|e| format!("Can't serialize Demand properties. Error: {}", e))?;
+        let offer_properties = serde_json::from_str(&self.offer_properties)
+            .map_err(|e| format!("Can't serialize Offer properties. Error: {}", e))?;
+
+        let demand = Demand {
+            properties: demand_properties,
+            constraints: self.demand_constraints,
+            demand_id: Some(self.demand_id.to_string()),
+            requestor_id: Some(self.requestor_id.to_string()),
+        };
+        let offer = Offer {
+            properties: offer_properties,
+            constraints: self.offer_constraints,
+            offer_id: Some(self.offer_id.to_string()),
+            provider_id: Some(self.provider_id.to_string()),
+        };
+        Ok(ClientAgreement {
+            agreement_id: self.id.to_string(),
+            demand,
+            offer,
+            valid_to: DateTime::<Utc>::from_utc(self.valid_to, Utc),
+            approved_date: self
+                .approved_date
+                .map(|d| DateTime::<Utc>::from_utc(d, Utc)),
+            state: self.state.into(),
+            proposed_signature: self.proposed_signature,
+            approved_signature: self.approved_signature,
+            committed_signature: self.committed_signature,
+        })
+    }
+}
```

Comment:
We wrote the same with @jiivan in get gsb agreement :(

Reply:
:(